### PR TITLE
doxygen : Add header file info for network libs APIs

### PR DESCRIPTION
**Pull request by heejin-kim**
You can see which path should be included to use network libs APIs.

### CODE REVIEW

**Comment by jeongchanKim** [comment]
Please check and add "SYSTEM CALL API" description in @details.

**Comment by sunghan-chang** [comment]
@heejin-kim Could you check @jeongchanKim 's comment?

**Comment by heejin-kim** [comment]
@sunghan-chang Yes, I'm doing. Also to add POSIX or not, it takes some time. I will be hurry.

**Comment by heejin-kim** [comment]
@sunghan-chang Please review

File: os/include/netdb.h [hunk 1]
```diff
@@ -331,6 +339,9 @@ void freeaddrinfo(FAR struct addrinfo *ai);
 
 /**
  * @brief gethostbyaddr() is the function returns a corresponding hostname with given IP addresses
+ *
+ * @details @b #include <netdb.h>\n
+ * SYSTEM CALL API
```
Comment (jeongchanKim) [inline os/include/netdb.h]: please check if this API is SYSTEM CALL or not.

File: os/include/netdb.h [hunk 2]
```diff
@@ -339,6 +350,9 @@ FAR struct hostent *gethostbyaddr(FAR const void *addr, socklen_t len, int type)
 
 /**
  * @brief gethostbyname() is the function sending DNS query to DNS server with hostname
+ *
+ * @details @b #include <netdb.h>\n
+ * SYSTEM CALL API
```
Comment (jeongchanKim) [inline os/include/netdb.h]: please check if this API is SYSTEM CALL or not.

File: os/include/sys/socket.h [hunk 3]
```diff
@@ -243,6 +279,9 @@ int getsockopt(int sockfd, int level, int option, FAR void *value, FAR socklen_t
 /**
 * @brief  get the socket name
 *
+* @details @b #include <sys/socket.h>\n
+* SYSTEM CALL API\n
```
Comment (jeongchanKim) [inline os/include/sys/socket.h]: please check if this API is SYSTEM CALL or not.

File: os/include/sys/socket.h [hunk 4]
```diff
@@ -254,6 +293,9 @@ int getsockname(int sockfd, FAR struct sockaddr *addr, FAR socklen_t *addrlen);
 /**
 * @brief  get the name of the peer socket
 *
+* @details @b #include <sys/socket.h>\n
+* SYSTEM CALL API\n
```
Comment (jeongchanKim) [inline os/include/sys/socket.h]: please check if this API is SYSTEM CALL or not.

File: os/include/arpa/inet.h [hunk 5]
```diff
@@ -162,6 +162,9 @@ extern "C"
 /**
  * @brief Convert the unsigned integer netlong from network byte order to host byte order
  *
+ * @details @b #include <arpa/inet.h>\n
+ * SYSTEM CALL API\n
```
Comment (jeongchanKim) [inline os/include/arpa/inet.h]: please check if this API is SYSTEM CALL or not.

File: os/include/arpa/inet.h [hunk 6]
```diff
@@ -171,6 +174,9 @@ uint32_t ntohl(uint32_t nl);
 /**
  * @brief Convert the unsigned short integer netshort from network byte order to host byte order
  *
+ * @details @b #include <arpa/inet.h>\n
+ * SYSTEM CALL API\n
```
Comment (jeongchanKim) [inline os/include/arpa/inet.h]: please check if this API is SYSTEM CALL or not.

File: os/include/arpa/inet.h [hunk 7]
```diff
@@ -180,6 +186,9 @@ uint16_t ntohs(uint16_t ns);
 /**
  * @brief Convert the unsigned integer hostlong from host byte order to network byte order
  *
+ * @details @b #include <arpa/inet.h>\n
+ * SYSTEM CALL API\n
```
Comment (jeongchanKim) [inline os/include/arpa/inet.h]: please check if this API is SYSTEM CALL or not.

File: os/include/arpa/inet.h [hunk 8]
```diff
@@ -189,6 +198,9 @@ uint32_t htonl(uint32_t hl);
 /**
  * @brief Convert the unsigned short integer hostshort from host byte order to network byte order
  *
+ * @details @b #include <arpa/inet.h>\n
+ * SYSTEM CALL API\n
```
Comment (jeongchanKim) [inline os/include/arpa/inet.h]: please check if this API is SYSTEM CALL or not.

File: os/include/arpa/inet.h [hunk 9]
```diff
@@ -199,6 +211,9 @@ uint16_t htons(uint16_t hs);
 /**
  * @brief Convert the Internet host address from the IPv4 numbers-and-dots notation into binary form
  *
+ * @details @b #include <arpa/inet.h>\n
+ * SYSTEM CALL API\n
```
Comment (jeongchanKim) [inline os/include/arpa/inet.h]: please check if this API is SYSTEM CALL or not.

File: os/include/arpa/inet.h [hunk 10]
```diff
@@ -209,6 +224,9 @@ int         inet_aton(FAR const char *cp, FAR struct in_addr *inp);
 /**
  * @brief Convert the Internet host address from the IPv4 numbers-and-dots notation into binary data in network byte order.
  *
+ * @details @b #include <arpa/inet.h>\n
+ * SYSTEM CALL API\n
```
Comment (jeongchanKim) [inline os/include/arpa/inet.h]: please check if this API is SYSTEM CALL or not.

File: os/include/arpa/inet.h [hunk 11]
```diff
@@ -223,6 +241,9 @@ in_addr_t   inet_network(FAR const char *cp);
 /**
  * @brief Convert the Internet host address to a string in IPv4 dotted-decimal notation.
  *
+ * @details @b #include <arpa/inet.h>\n
+ * SYSTEM CALL API\n
```
Comment (jeongchanKim) [inline os/include/arpa/inet.h]: please check if this API is SYSTEM CALL or not.

File: os/include/netdb.h [hunk 12]
```diff
@@ -312,6 +312,10 @@ void endservent(void);
 
 /**
  * @brief getaddrinfo() is a function that returns information on a particular host name (such as its IP address) and loads up a struct sockaddr.
+ *
+ * @details @b #include <netdb.h>\n
+ * SYSTEM CALL API\n
```
Comment (jeongchanKim) [inline os/include/netdb.h]: please check if this API is SYSTEM CALL or not.

File: os/include/netdb.h [hunk 13]
```diff
@@ -323,6 +327,10 @@ int getaddrinfo(const char *host, const char *service, const struct addrinfo *hi
 
 /**
  * @brief freeaddrinfo() frees the memory allocated by the getaddrinfo() function.
+ *
+ * @details @b #include <netdb.h>\n
+ * SYSTEM CALL API\n
```
Comment (jeongchanKim) [inline os/include/netdb.h]: please check if this API is SYSTEM CALL or not.

File: os/include/arpa/inet.h [hunk 14]
```diff
@@ -199,6 +211,9 @@ uint16_t htons(uint16_t hs);
 /**
  * @brief Convert the Internet host address from the IPv4 numbers-and-dots notation into binary form
  *
+ * @details @b #include <arpa/inet.h>\n
+ * SYSTEM CALL API\n
+ * POSIX API (refer to : http://pubs.opengroup.org/onlinepubs/9699919799/)
```
Comment (jeongchanKim) [inline os/include/arpa/inet.h]: please check if this API is POSIX or not.

Comment (heejin-kim) [reply os/include/arpa/inet.h]: I found that just like below.

Header : <arpa/inet.h>

in_addr_t    inet_addr(const char *);
char        *inet_ntoa(struct in_addr);
const char  *inet_ntop(int, const void *restrict, char *restrict, socklen_t);
int          inet_pton(int, const char *restrict, void *restrict);

If it is not the POSIX API, please let me know. I will revise that.

File: os/include/arpa/inet.h (hunk 5 above)
Comment (heejin-kim) [reply os/include/arpa/inet.h]: revised

File: os/include/arpa/inet.h (hunk 6 above)
Comment (heejin-kim) [reply os/include/arpa/inet.h]: revised

File: os/include/arpa/inet.h (hunk 7 above)
Comment (heejin-kim) [reply os/include/arpa/inet.h]: revised

File: os/include/arpa/inet.h (hunk 8 above)
Comment (heejin-kim) [reply os/include/arpa/inet.h]: revised

File: os/include/arpa/inet.h (hunk 9 above)
Comment (heejin-kim) [reply os/include/arpa/inet.h]: revised

File: os/include/arpa/inet.h (hunk 10 above)
Comment (heejin-kim) [reply os/include/arpa/inet.h]: revised

File: os/include/arpa/inet.h (hunk 11 above)
Comment (heejin-kim) [reply os/include/arpa/inet.h]: revised

File: os/include/netdb.h (hunk 12 above)
Comment (heejin-kim) [reply os/include/netdb.h]: revised

File: os/include/netdb.h (hunk 13 above)
Comment (heejin-kim) [reply os/include/netdb.h]: revised

File: os/include/netdb.h (hunk 1 above)
Comment (heejin-kim) [reply os/include/netdb.h]: revised

File: os/include/netdb.h (hunk 2 above)
Comment (heejin-kim) [reply os/include/netdb.h]: revised

File: os/include/sys/socket.h (hunk 3 above)
Comment (heejin-kim) [reply os/include/sys/socket.h]: revised

File: os/include/sys/socket.h (hunk 4 above)
Comment (heejin-kim) [reply os/include/sys/socket.h]: revised

**Pull request merged**